Enh/rearrange components structure

louvrecly opened the pull request:
### Descriptions
- Restructure components and rewrite route logic for easier debugging

### Stories
- [x] Move `DrawerContainer` into `DrawerContainer` folder
- [x] Extract the contents of bottom drawer to create `BottomDrawerContents` from `Main`
- [x] Move `Map` into `Main` and Move `loadThreads`, `loadUsers` and `setRecorders` props into `Main`
- [x] Rewrite `App` to contain `Switch` and `Routes` only
- [x] Add `location` effect hook and `loading` state in `Main`
- [x] Create `ThreadRoute` component to check for valid `threadId` in route param
- [x] Refactor `BottomDrawerContents` and Add cleanup effect to navigate back to main page
- [x] Add on render effect in `ThreadPanel`
- [x] Refactor `TimerBar` to use updating `centisecond` effect with cleanup to `clearInterval` and set `recording` state to `false`
- [x] Create `EmbeddedRecordButton`
- [x] Create `RecordButtonBloc`
- [x] Refactor `RecordButton` to add `captureAudio` and `useDidUpdateEffect` hook to handle `isRecording` state update
- [x] Refactor `VoiceForm` to add `loading` state effect hook and `EmbeddedRecordButton`
- [x] Create `NewVoice` to handle `/new` route display logic
- [x] Clean up `DrawerContainer` logic
- [x] Refactor `AudioPlayer` to remove `setShowRecordButtonState` and `embedRecordButton` methods
- [x] Create `ThreadRouteParam` interface
- [x] Refactor `Map` to remove `match` monitoring effect
- [x] Refactor `geolocation` util to use `watchPosition` instead of `getCurrentPosition`
- [x] Remove `embedRecordButton` action
- [x] Rearrange dependencies in `package.json`

thrinithomas commented:
Impressive work! Just one thing related to side drawer. After I pressed the hamburger menu, the side drawer pop out, and the 9UP button disappear. But the button won't display after side drawer is disappear.

louvrecly commented:
> Impressive work! Just one thing related to side drawer. After I pressed the hamburger menu, the side drawer pop out, and the 9UP button disappear. But the button won't display after side drawer is disappear.

removed the [hide `RecordButton` logic in `HeadNav`](https://github.com/louvrecly/ab-test-web/pull/41/files#diff-ae19119aa4a95ce80921c6f3d84682d4b335de95e80f5d850514abd9ca005480R32-R35)
added a `LeftDrawerContents` components to control the show / hide `RecordButton` logic in [this effect hook](https://github.com/louvrecly/ab-test-web/pull/41/files#diff-272dd851c4dfb3bfed8df931cd17e74f3855adc790a46da6c88b2da7720c675dR14-R18)
